docs/src/bot.md

```diff
@@ -42,95 +48,49 @@ snoop_bot(
-```julia
-println("Benchmarking the inference time of the tests")
-snoop_bench(BotConfig("MatLang"))
-```
+Finally, you could use package loading as the only operation,
+with `snoop_bot(config, :(using MyPackage))`.
 
-To selectively exclude some of your tests from running by SnoopCompile bot, use the global SnoopCompile_ENV::Bool variable.
-```julia
-if !isdefined(Main, :SnoopCompile_ENV) || SnoopCompile_ENV == false
-  # the tests you want to skip in SnoopCompile environment
-end
-```
+`snoop_bot` uses different strategies depending on the Julia version:
 
-Benchmarking inference time of loading
-```julia
-println("Benchmarking inference time of loading")
-snoop_bench( BotConfig("MatLang"), :(using MatLang) ) # this syntax should be avoided for complex expressions
-```
+- On Julia 1.2 or higher, it identifies methods for precompilation based on [`@snoopi`](@ref macro-snoopi);
+- On Julia 1.0 or 1.1 (which do not support `@snoopi`), it identifies methods for precompilation based on [`@snoopc`](@ref macro-snoopc).
 
-[Zygote example](https://github.com/aminya/Zygote.jl/blob/SnoopCompile/deps/SnoopCompile/snoop_bench.jl)
+You can override this default behavior with a keyword argument, see [`snoop_bot`](@ref) for details.
 
-`snoop_bench` has three variants:
-- `snoopi_bench`: benchmarks the inference time.
-  This is the default behavior of `snoop_bench` for Julia >= 1.2
-- `snoopv_bench`: tracks the time using `@timev`. This is the default behavior of `snoop_bench` for Julia < 1.2
+## 4 - Optionally test the impact of your precompiles with `snoop_bench`
 
+Call [`snoop_bench`](@ref) to measure the effect of adding precompile files.
+It takes the same parameters as `snoop_bot` above.
+You can run this manually, or choose to run it during automatic precompile file generation.
+To perform it automatically, create a `snoop_bench.jl` script in the bot configuration directory.
+This should be nearly identical to your `snoop_bot.jl` file, but calling `snoop_bench`
+instead.
+Note that benchmarking has the option of different performance metrics,
+`snoop_mode=:snoopi` or `snoop_mode=:run_time` depending on whether you want to measure inference time or the run time of your precompile script.
 
-## 4 - GitHub Action file (only for online run)
+## 5 - Configure the bot to run with a GitHub Action file
 
-In your repository, create a workflow file under `.github/workflows/SnoopCompile.yml`, and use the following content:
+You can create the precompile files automatically when you merge pull requests to `master` by adding a workflow file under `.github/workflows/SnoopCompile.yml`.
+This file should have content such as the example below.
+Lines marked with `NOTE` deserve special attention as likely places you may
+need to adjust the configuration.
 
 ```yaml
 name: SnoopCompile
 
-
-# Edit based on your repository.
 on:
   push:
     branches:
-      # - 'master'
-  # pull_request:  # comment for big repositories
+      - 'master'
```

Comment:
Right?

Reply:
I prefer having `master` be commented, and only used for big repos. Personally, I usually, create branches for developing features, and I like to see the bot working on those branches as well. I can add a short comment about this.